packages/devtools_app/lib/src/performance/performance_screen.dart

```diff
@@ -360,6 +363,45 @@ class _SecondaryControls extends StatelessWidget {
   }
 }
 
+class DebuggingOptionsDialog extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    return DevToolsDialog(
+      title: dialogTitleText(theme, 'Debugging Options'),
+      includeDivider: false,
+      content: Container(
+        width: defaultDialogWidth,
+        child: Column(
+          mainAxisSize: MainAxisSize.min,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            _maybeIncludeOption(performanceOverlay),
+            _maybeIncludeOption(profileWidgetBuilds),
+            // TODO(devoncarew): Enable this once we have a UI displaying the
+            // values.
+            // _maybeIncludeOption(trackRebuildWidgets),
+            _maybeIncludeOption(disableClipLayers),
+            _maybeIncludeOption(disableOpacityLayers),
+            _maybeIncludeOption(disablePhysicalShapeLayers),
+          ],
+        ),
+      ),
+      actions: [
+        DialogCloseButton(),
+      ],
+    );
+  }
+
+  Widget _maybeIncludeOption(ToggleableServiceExtensionDescription service) {
+    if (serviceManager.serviceExtensionManager
```

Comment:
I would instead make ServiceExtensionCheckbox itself show up as disabled or render as a SizedBox rather than adding the check here. 
Doing this here makes the screen fragile as if you happened to open the dialog right when the app is starting up or right after a hot restart, the checkboxes won't show up unless you close the dialog and reopen. If the logic is tied to the checkbox itself, it can listen for when the service extension becomes available to change its state as appropriate.

Reply:
done